src/index.js

```diff
@@ -0,0 +1,277 @@
+const tangramReference = require('tangram-reference').load();
+const color = require('./color.js');
+
+//getReferenceIndexedByCSS returns a reference based on ref indexed by the CSS property name instead of the reference original property name
+//it does a shallow copy of each property, therefore, it's possible to change the returned reference by changing the original one, don't do it
+function getReferenceIndexedByCSS(ref) {
+    var newRef = {};
+    for (var symb in ref.symbolizers) {
+        for (var property in ref.symbolizers[symb]) {
+            newRef[ref.symbolizers[symb][property].css] = ref.symbolizers[symb][property];
+        }
+    }
+    return newRef;
+}
+const referenceCSS = getReferenceIndexedByCSS(tangramReference);
+
+
+function wrapFn(functionBody) {
+    return 'function (){var data=feature; var ctx={zoom: $zoom};' + functionBody + '}';
+}
+function getReferenceDefault(property) {
+    return referenceCSS[property]['default-value'];
+}
+
+function getLiteralFromShaderValue(shaderValue) {
+    // jshint ignore:start
+    // required within the eval context
+    const ctx = { zoom: 10 };
+    // jshint ignore:end
+    var _value = null;
+    shaderValue.js.forEach(function (code) {
+        eval(code);
+    });
+    return _value;
+}
+
+//Returns the final opacity override selecting between fill-opacity and outline-opacity.
```

Comment:
I would consider using [JSDoc](http://usejsdoc.org/). What do you think?

Reply:
I don't have experience with it, but I could use it. However, should we hold this PR for this?

Reply:
No, no need to hold it. I just wanted to raise some awareness :).